### PR TITLE
JW-943 Merge jobs

### PR DESCRIPTION
**Pull request by m-kostrzewa**

### CODE REVIEW

File: CIScripts/BuildDockerDriver.ps1 [hunk 1]
```diff
@@ -1,23 +1,13 @@
...
 New-Item -ItemType Directory ./bin
-Set-Location bin
+Push-Location bin
 
 Write-Host "Installing test runner"
 go get -u -v github.com/onsi/ginkgo/ginkgo
 
 Write-Host "Building driver"
-go build -v $Env:DRIVER_SRC_PATH
+go get -u -v $Env:DRIVER_SRC_PATH
```
Comment (m-kostrzewa) [inline CIScripts/BuildDockerDriver.ps1]: I think this actually pulls from https://github.com/codilime/contrail-windows-docker. Furthermore, I think this builds only the master branch.

I suspect that only `go build` can be used here...

File: CIScripts/BuildFunctions.ps1 [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    
+    Write-Host "Copying Agent dependencies"
+    Copy-Item -Recurse "$ThirdPartyCache\agent\*" third_party/
+
+    
+    Write-Host "Building Agent, MSI and API"
+    scons contrail-vrouter-agent contrail-vrouter-agent.msi controller/src/vnsw/contrail_vrouter_api:sdist
+    if ($LASTEXITCODE -ne 0) {
+    throw "Building Agent failed"
+}
```
Comment (jablonskim) [inline CIScripts/BuildFunctions.ps1]: Wrong indentation in this 2 lines

File: CIScripts/BuildFunctions.ps1 [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    Write-Host "Copying Extension dependencies"
+    Copy-Item -Recurse "$ThirdPartyCache\extension\*" third_party\
+    Copy-Item -Recurse third_party\cmocka vrouter\test\
+
+    
+    Write-Host "Building Extension and Utils"
+    scons vrouter
+    if ($LASTEXITCODE -ne 0) {
+    throw "Building vRouter solution failed"
+}
```
Comment (jablonskim) [inline CIScripts/BuildFunctions.ps1]: Wrong indentation in this 2 lines

File: CIScripts/BuildFunctions.ps1 [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    Param ([Parameter(Mandatory = $true)] [string] $SigntoolPath,
+           [Parameter(Mandatory = $true)] [string] $CertPath,
+           [Parameter(Mandatory = $true)] [string] $CertPasswordFilePath,
+           [Parameter(Mandatory = $true)] [string] $MSIPath)
+    
+    $cerp = Get-Content $CertPasswordFilePath
+    & $SigntoolPath sign /f $CertPath /p $cerp $MSIPath
+    if ($LASTEXITCODE -ne 0) {
+    throw "Signing $MSIPath failed"
+}
```
Comment (jablonskim) [inline CIScripts/BuildFunctions.ps1]: Wrong indentation in lines above

File: CIScripts/BuildFunctions.ps1 [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+    Repo ([string] $Url, [string] $Branch, [string] $Dir, [string] $DefaultBranch) {
+        $this.Url = $Url
+        $this.Branch = $Branch
+        $this.Dir = $Dir
+        $this.DefaultBranch = $DefaultBranch
+    }
+}
+
+function Clone-Repos {
```
Comment (jablonskim) [inline CIScripts/BuildFunctions.ps1]: Please use approved words in function names (https://msdn.microsoft.com/en-us/library/ms714428(v=vs.85).aspx)

File: CIScripts/BuildFunctions.ps1 [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+            git clone -b $_.Branch $_.Url $_.Dir
+
+            if ($LASTEXITCODE -ne 0) {
+                throw "Cloning from " + $_.Url + " failed"
+            }
+        }
+    })
+}
+
+function Contrail-Common-Actions {
```
Comment (jablonskim) [inline CIScripts/BuildFunctions.ps1]: Please use approved words in function names (https://msdn.microsoft.com/en-us/library/ms714428(v=vs.85).aspx)

File: CIScripts/BuildFunctions.ps1 [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+    Invoke-BatchFile "$VSSetupEnvScriptPath"
+    
+    Write-Host "Copying common third-party dependencies"
+    New-Item -ItemType Directory .\third_party
+    Copy-Item -Recurse "$ThirdPartyCache\common\*" third_party\
+
+    Copy-Item tools\build\SConstruct .\
+}
+
+function Sign-MSI {
```
Comment (jablonskim) [inline CIScripts/BuildFunctions.ps1]: Please use approved words in function names (https://msdn.microsoft.com/en-us/library/ms714428(v=vs.85).aspx)

File: CIScripts/BuildFunctions.ps1 [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+           [Parameter(Mandatory = $true)] [string] $MSIPath)
+    
+    $cerp = Get-Content $CertPasswordFilePath
+    & $SigntoolPath sign /f $CertPath /p $cerp $MSIPath
+    if ($LASTEXITCODE -ne 0) {
+    throw "Signing $MSIPath failed"
+}
+}
+
+function Build-DockerDriver {
```
Comment (jablonskim) [inline CIScripts/BuildFunctions.ps1]: Please use approved words in function names (https://msdn.microsoft.com/en-us/library/ms714428(v=vs.85).aspx)

File: CIScripts/BuildFunctions.ps1 [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+    Pop-Location
+
+    Move-Item $srcPath/installer.msi ./
+    
+    Sign-MSI -SigntoolPath $SigntoolPath -CertPath $CertPath -CertPasswordFilePath $CertPasswordFilePath -MSIPath "installer.msi"
+
+    Pop-Location
+}
+
+function Build-Extension {
```
Comment (jablonskim) [inline CIScripts/BuildFunctions.ps1]: Please use approved words in function names (https://msdn.microsoft.com/en-us/library/ms714428(v=vs.85).aspx)

File: CIScripts/BuildFunctions.ps1 [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+    $utilsMSI = "build\debug\vrouter\utils\utils.msi"
+    
+    Write-Host "Signing utilsMSI"
+    Sign-MSI -SigntoolPath $SigntoolPath -CertPath $CertPath -CertPasswordFilePath $CertPasswordFilePath -MSIPath $utilsMSI
+    
+    Write-Host "Signing vRouterMSI"
+    Sign-MSI -SigntoolPath $SigntoolPath -CertPath $CertPath -CertPasswordFilePath $CertPasswordFilePath -MSIPath $vRouterMSI
+}
+
+function Build-Agent {
```
Comment (jablonskim) [inline CIScripts/BuildFunctions.ps1]: Please use approved words in function names (https://msdn.microsoft.com/en-us/library/ms714428(v=vs.85).aspx)

**Review by jablonskim — CHANGES_REQUESTED** [review]

File: CIScripts/BuildAll.ps1 [hunk 11]
```diff
@@ -0,0 +1,20 @@
+. $PSScriptRoot\InitializeCIScript.ps1
+. $PSScriptRoot\BuildFunctions.ps1
+
+$Repos = @(
+    [Repo]::new($Env:DRIVER_REPO_URL, $Env:DRIVER_BRANCH, "src/github.com/codilime/contrail-windows-docker" ,"master"),
```
Comment (jablonskim) [inline CIScripts/BuildAll.ps1]: Small formatting issue here (`" ,"` => `", "`).

File: CIScripts/BuildDockerDriver.ps1 [hunk 12]
```diff
@@ -1,46 +1,10 @@
-if ($Env:ENABLE_TRACE -eq $true) {
-    Set-PSDebug -Trace 1
-}
+. $PSScriptRoot\InitializeCIScript.ps1
+. $PSScriptRoot\BuildFunctions.ps1
 
-# Refresh Path
-$Env:Path = [System.Environment]::GetEnvironmentVariable("Path", "Machine")
+$Repos = @(
+    [Repo]::new($Env:DRIVER_REPO_URL, $Env:DRIVER_BRANCH, "src/github.com/codilime/contrail-windows-docker" ,"master")
```
Comment (jablonskim) [inline CIScripts/BuildDockerDriver.ps1]: Small formatting issue here (`" ,"` => `", "`).

**Review by jablonskim — DISMISSED** [review]

**Review by jablonskim — DISMISSED** [review]

**Review by jablonskim — DISMISSED** [review]

File: CIScripts/BuildFunctions.ps1 [hunk 13]
```diff
@@ -0,0 +1,151 @@
+class Repo {
+    [string] $Url;
+    [string] $Branch;
+    [string] $DefaultBranch;
+    [string] $Dir;
+
+    Repo ([string] $Url, [string] $Branch, [string] $Dir, [string] $DefaultBranch) {
```
Comment (Attil) [inline CIScripts/BuildFunctions.ps1]: I think changing the order of parameters here to be consistent with the order of class fields would make the code easier to read. Or the other way around.

File: CIScripts/BuildFunctions.ps1 [hunk 14]
```diff
@@ -0,0 +1,151 @@
...
+           [Parameter(Mandatory = $true)] [string] $CertPath,
+           [Parameter(Mandatory = $true)] [string] $CertPasswordFilePath)
+    
+    $Env:GOPATH=pwd
+    
+    New-Item -ItemType Directory ./bin
+    Push-Location bin
+
+    Write-Host "Installing test runner"
+    go get -u -v github.com/onsi/ginkgo/ginkgo
```
Comment (Attil) [inline CIScripts/BuildFunctions.ps1:78]: Are you sure we should always use the latest version of ginkgo?

File: CIScripts/BuildFunctions.ps1 [hunk 15]
```diff
@@ -0,0 +1,151 @@
...
+    $modules.ForEach({
+        .\ginkgo.exe build $srcPath/$_
+        Move-Item $srcPath/$_/$_.test ./
+    })
+
+    Write-Host "Copying Agent API python script"
+    Copy-Item $srcPath/scripts/agent_api.py ./
+
+    Write-Host "Intalling MSI builder"
+    go get -u -v github.com/mh-cbon/go-msi
```
Comment (Attil) [inline CIScripts/BuildFunctions.ps1:97]: Are you sure we should always use the latest version of Go MSI?

**Review by Attil — CHANGES_REQUESTED** [review]

File: CIScripts/BuildFunctions.ps1 (hunk 14 above)
Comment (katrybacka) [reply CIScripts/BuildFunctions.ps1:78]: Yes

File: CIScripts/BuildFunctions.ps1 (hunk 15 above)
Comment (katrybacka) [reply CIScripts/BuildFunctions.ps1:97]: Yes

**Review by Attil — APPROVED** [review]

**Review by jablonskim — APPROVED** [review]

**Pull request merged**